depthimage_to_navscan_rgbd/src/depthimage_to_navscan_rgbd.cpp

```diff
@@ -76,12 +83,18 @@ int main(int argc, char** argv)
 
     ros::Publisher pointcloud2_publisher = nh.advertise<sensor_msgs::PointCloud2>("navscan", 20);
 
+    ros::Rate r(rate);
+    ros::Rate r_inner(20*rate);
     while (ros::ok())
     {
         rgbd::ImageConstPtr image;
         geo::Pose3D sensor_pose;
-        if (!image_buffer.nextImage(image, sensor_pose))
+        r_inner.reset();
+        while (!image_buffer.nextImage(image, sensor_pose))
+        {
+            r_inner.sleep();
```

Comment:
use image_buffer.waitForRecentImage if you are willing to wait for a new image

Reply:
`waitForRecentImage` waits for 0.1s till timeout. So that one is dirtier than this one. 😉 
As the waiting time should scale with the timeout.

Reply:
The image buffer should be responsible for deciding how long a waittime is appropriate since that component is connected to the image feed. Right now we have duplicated functionalityof the image buffer in this module.

Reply:
As this is very early in the loop. The `waitForRecentImage` could be used here and still get an consistent loop rate. Under the condition we change the waiting in that function.

P.S. wait is done on many points. So this is duplicate everywhere 😉

Reply:
I think this is duplicating the functionality of `waitForRecentImage`, this PR is not a clean solution in my opinion